src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCheck.java

```diff
@@ -109,11 +109,11 @@ public void visitToken(DetailAST ast) {
         final String line = getLine(ast.getLineNo() - 1);
         final int before = ast.getColumnNo() - 1;
 
-        if ((before == -1 || Character.isWhitespace(line.charAt(before)))
-                && !isInEmptyForInitializerOrCondition(ast)) {
+        if (!isInEmptyForInitializerOrCondition(ast)
+                && (before == -1 || Character.isWhitespace(line.charAt(before)))) {
             boolean flag = !allowLineBreaks;
             // verify all characters before '.' are whitespace
-            for (int i = 0; !flag && i <= before - 1; i++) {
```

Comment:
`flag` can be killed because we break the loop when it is turned true so this condition is pointless.